performance_test/helper_scripts/performance_test_file_reader.py

```diff
@@ -7,45 +7,71 @@
-else:
-    print("Usage: python performance_test_file_reader.py /path_to_log_files")
-    sys.exit(0)
-
-logfiles = []
-N = 15  # Number of line to skip before CSV data starts.
 
-for file in os.listdir(directory):
-    if file.startswith("log_"):
-        print(os.path.join(directory, file))
-        logfiles.append(os.path.join(directory, file))
-
-for f in logfiles:
+def parse_file(file, single_file=None):
+    if not os.path.basename(file).startswith("log_") or file.endswith(".pdf"):
+        print("Skipping "+str(file))
+        return
     try:
-        print("Parsing file:" + str(f))
-        dataframe = pd.read_csv(f, skiprows=N + 1, sep="[ \t]*,[ \t]*", engine='python')
-        # dataframe = dataframe.drop(columns=['Unnamed: 19'])
-        with open(f) as myfile:
+        print("Parsing file:" + str(file))
+        dataframe = pd.read_csv(file, skiprows=N + 1, sep="[ \t]*,[ \t]*", engine='python')
+        with open(file) as myfile:
             head = [next(myfile) for x in range(0, N)]
-            print(''.join(head))
 
         if not dataframe.empty:
             pd.options.display.float_format = '{:.4f}'.format
+            dataframe = dataframe.iloc[10:]
+            dataframe["maxrss"] = dataframe["ru_maxrss"] / 1e3
             dataframe.drop(list(dataframe.filter(regex='ru_')), axis=1, inplace=True)
             dataframe["latency_variance (ms) * 100"] = 100.0 * dataframe["latency_variance (ms)"]
+            # ru_maxrss is in KB: http://man7.org/linux/man-pages/man2/getrusage.2.html. Converting to Mb
+            dataframe["maxrss (Mb)"] = dataframe["maxrss"] / 1e3
             dataframe[["T_experiment",
                        "latency_min (ms)",
                        "latency_max (ms)",
                        "latency_mean (ms)",
-                       "latency_variance (ms) * 100"]] \
-                .plot(x='T_experiment')
+                       "latency_variance (ms) * 100",
+                        "maxrss (Mb)"]] \
+                .plot(x='T_experiment', secondary_y=["maxrss (Mb)"])
+
             plt.figtext(0.0, 1.0, ''.join(head), fontsize=8, horizontalalignment='left')
             plt.figtext(0.65, 0.9, dataframe.mean().round(4), fontsize=8,
                         horizontalalignment='left')
-            plt.savefig(os.path.basename(f) + ".pdf",
-                        bbox_inches=matplotlib.transforms.Bbox(np.array(((0, 0), (8, 8)))))
+
+            if single_file is None:
+                plt.savefig(file + ".pdf",
+                            bbox_inches=matplotlib.transforms.Bbox(np.array(((0, 0), (8, 8)))))
+            else:
+                single_file.savefig(bbox_inches=matplotlib.transforms.Bbox(np.array(((0, 0), (8, 8)))))
+            plt.close('all')
     except:  # noqa: E722 I do rethrow.
-        print("Could not parse file: " + str(f) + "\n")
-        raise
+        print("Could not parse file: " + str(file) + "\n")
+
+
+pdf = None
+if len(sys.argv) == 2:
+    directory = sys.argv[1]
```

Comment:
If the `single_pdf_output_file` argument is not provided will the result be stored in a pdf file of the log name?

Reply:
Yes, my idea was that a single file output is designed to merge the plots from multiple log files into a single pdf which makes no sense if there is only one log file. On the other hand, this might be confusing for the user, so I changed it to make things consistent.

Reply:
agreed